cleanup unawaited futures lints

danrubel opened the pull request:
@abarth

Commit:
cleanup unawaited futures lints

Hixie commented on packages/flutter/test/widget/bottom_sheet_test.dart:
```diff
@@ -23,6 +23,7 @@ void main() {
     await tester.pump();
     expect(find.text('BottomSheet'), findsNothing);
 
+    // ignore: unawaited_futures
```
rather than ignoring the future, we should capture it, and verify that it completes when appropriate

danrubel replied:
Good idea. Done.

Hixie commented on packages/flutter/test/widget/bottom_sheet_test.dart:
```diff
@@ -71,6 +73,7 @@ void main() {
     expect(showBottomSheetThenCalled, isFalse);
     expect(find.text('BottomSheet'), findsNothing);
 
+    // ignore: unawaited_futures
```
ditto

danrubel replied:
Done.

danrubel replied:
This one does not need to be changed. It is already asserting that the returned future completes as expected.

Hixie commented on packages/flutter/test/widget/positioned_test.dart:
```diff
@@ -53,6 +53,7 @@ void main() {
       )
     ); // t=0
     recordMetrics();
+    // ignore: unawaited_futures
```
ditto

danrubel replied:
Done.

Hixie commented on packages/flutter/test/widget/remember_scroll_position_test.dart:
```diff
@@ -57,6 +57,7 @@ Future<Null> performTest(WidgetTester tester, bool maintainState) async {
   expect(find.text('10'), findsNothing);
   expect(find.text('100'), findsNothing);
 
+  // ignore: unawaited_futures
```
ditto

danrubel replied:
Done.

danrubel replied:
Done.

danrubel replied:
Done.

Hixie commented:
I think for all of these we should capture the futures, and verify that they complete when appropriate.

Commit:
Merge branch 'master' into unawaited-futures-3

Commit:
revised tests to assert futures complete as expected

danrubel commented on packages/flutter/test/widget/bottom_sheet_test.dart:
```diff
@@ -44,6 +45,7 @@ void main() {
     await tester.pump(new Duration(seconds: 1)); // frame after the animation (sheet has been removed)
     expect(find.text('BottomSheet'), findsNothing);
 
+    // ignore: unawaited_futures
```
Done.

danrubel commented:
I've overhauled the PR to revised tests to assert futures complete as expected.
PTAL @Hixie

Hixie commented on packages/flutter/lib/src/widgets/scrollable.dart:
```diff
@@ -441,6 +441,8 @@ class ScrollableState<T extends Scrollable> extends State<T> {
   /// the new offset, do not provide a duration, rather than providing
   /// a zero duration.
+  ///
+  /// The returned [Future] completes when scrolling animation is complete.
```
when the

danrubel replied:
Fixed.

Hixie commented on packages/flutter/lib/src/widgets/scrollable.dart:
```diff
@@ -519,6 +521,8 @@ class ScrollableState<T extends Scrollable> extends State<T> {
   /// offset with the given value as the initial velocity. The physics
   /// simulation is determined by the scroll behavior.
+  ///
+  /// The returned [Future] completes when scrolling animation is complete.
```
when the

danrubel replied:
Fixed.

Hixie commented on packages/flutter/test/widget/bottom_sheet_test.dart:
```diff
@@ -23,6 +23,8 @@ void main() {
     expect(find.text('BottomSheet'), findsNothing);
 
+    showBottomSheetThenCalled = false;
+    // ignore: unawaited_futures
```
wait, what's the future that isn't awaited? the future has a `.then()` and everything.

danrubel replied:
`.then` returns a Future so that it can be chained.

Hixie commented:
This is great stuff.

Hixie commented:
I'm concerned about the case where you have a future that has a `.then` with a completely synchronous body but the warning still triggers.

Hixie commented:
I'd like to better understand the warning before I use my LGTM stamp.

Commit:
fix dartdoc and remove ignore unawaited_future comments

danrubel commented:
Having the warning trigger even after I added a `.then` block is present bothered me as well, so I suggested a [change to the unawaited_future lint](https://github.com/dart-lang/sdk/issues/27401). In the meantime I have removed the `// ignore: unawaited_future` comments.
PTAL @Hixie

abarth commented on packages/flutter/lib/src/animation/animation_controller.dart:
```diff
@@ -154,6 +154,7 @@ class AnimationController extends Animation<double>
   AnimationStatus _status;
 
   /// Starts running this animation forwards (towards the end).
+  /// Returns a [Future] that completes when the animation is complete.
```
We usually have a blank line between the one-line summary and the details, so:

```
/// Starts running this animation forwards (towards the end).
///
/// Returns a [Future] that completes when the animation is complete.
```

danrubel replied:
Done.

abarth commented on packages/flutter/test/widget/bottom_sheet_test.dart:
```diff
@@ -23,6 +23,7 @@ void main() {
     await tester.pump();
     expect(find.text('BottomSheet'), findsNothing);
 
+    showBottomSheetThenCalled = false;
```
Why separate the declaration and initialization?  If you prefer, you can move the declaration here too.

danrubel replied:
Yes, I like that better. Done.

abarth commented on packages/flutter/test/widget/positioned_test.dart:
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
```
We usually put a blank line between the `dart:` imports and the `package:` imports.  I believe that's in some official Dart style guide somewhere.

danrubel replied:
Good catch. Done.

abarth commented on packages/flutter/test/widget/remember_scroll_position_test.dart:
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
```
ditto

danrubel replied:
Done.

abarth commented on packages/flutter/test/widget/scroll_events_test.dart:
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
```
ditto

danrubel replied:
Done.

abarth commented on packages/flutter/test/widget/scroll_events_test.dart:
```diff
@@ -22,6 +23,14 @@ Widget _buildScroller({Key key, List<String> log}) {
 void main() {
+  GlobalKey<ScrollableState<Scrollable>> scrollKey;
+
+  Completer<Null> scrollTo(double newScrollOffset, {Duration duration}) {
```
We put spaces after the `{` and before the `}` for named argument declarations because we're strange.

danrubel replied:
Ok. :-) Done.

abarth commented on packages/flutter/test/widget/scrollable_lazy_list_test.dart:
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
```
import style nit

danrubel replied:
Done.

abarth reviewed (approved):
LGTM

Commit:
Merge branch 'master' into unawaited-futures-3

Commit:
address comments
